src/libsync/syncengine.cpp

```diff
@@ -912,7 +912,16 @@ void SyncEngine::slotDiscoveryJobFinished(int discoveryResult)
             return;
         }
     }
-    if (!_hasForwardInTimeFiles && _backInTimeFiles >= 2) {
+
+    auto databaseFingerprint = _journal->dataFingerprint();
+    // If databaseFingerprint is null, this means that there was no information in the database
+    // (for example, upgrading from a previous version, or first sync)
+    // If it is empty (""), then that meanst that it was empty on the server before.
```

Comment:
I don't understand yet. Can't we just ignore it if it is empty or null? I don't think we can properly distinguish between empty and null in the PROPFIND property response can we?
(yes i know there is 404 properties but still).
I don't understand the reason to not just check for notEmpty?